snakemake/scheduler.py

```diff
@@ -419,7 +421,12 @@ def schedule(self):
                         "Ready jobs ({}):\n\t".format(len(needrun))
                         + "\n\t".join(map(str, needrun))
                     )
-                    run = self.job_selector(needrun)
+
+                    run = (
+                        self.job_selector(needrun, self.scheduler_ilp_solver)
+                        if self.scheduler_ilp_solver
+                        else self.job_selector(needrun)
+                    )
```

Comment:
Please keep this as before, and directly access self.scheduler_ilp_solver inside the job_selector_ilp method.